### PR TITLE
Add module verification

### PR DESCRIPTION
**Pull request by Gymbotan**

### CODE REVIEW

File: source/TuviProtonPrimitiveLib/Data/Constants.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+BwgDAgkQNQWFxOlRjyYEFQgKAgMWAgECGQECGwMCHgEAAPGRAP9sauJsW12U
+MnTQUZpsbJb53d0Wv55mZIIiJL2XulpWPQD/V6NglBd96lZKBmInSXX/kXat
+Sv+y0io+LR8i2+jV+AbOOARcAcuDEgorBgEEAZdVAQUBAQdAeJHUz1c9+KfE
+kSIgcBRE3WuXC4oj5a2/U3oASExGDW4DAQgHwmEEGBYIABMFAlwBy4MJEDUF
+hcTpUY8mAhsMAAD/XQD8DxNI6E78meodQI+wLsrKLeHn32iLvUqJbVDhfWSU
+WO4BAMcm1u02t4VKw++ttECPt+HUgPUq5pqQWe5Q2cW4TMsE
+=Y4Mw
+-----END PGP PUBLIC KEY BLOCK-----";
+
+        //internal static string SRP_MODULUS_KEY_FINGERPRINT => "248097092b458509c508dac0350585c4e9518f26";
```
Comment (al-kau) [inline source/TuviProtonPrimitiveLib/Data/Constants.cs:37]: remove SRP_MODULUS_KEY_FINGERPRINT

File: source/TuviAuthProtonLib/TuviAuthProtonLib.csproj [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk">
```
Comment (al-kau) [inline source/TuviAuthProtonLib/TuviAuthProtonLib.csproj:1]: What's changed here?

File: source/TuviAuthProtonLib/Broker.cs [hunk 3]
```diff
@@ -84,7 +84,7 @@ public async Task<Messages.Payloads.AuthResponse> AuthenticateAsync(string usern
                 username: username,
                 password: password,
                 salt: response.Salt,
-                modulus: response.Modulus,
+                signedMessage: response.Modulus,
```
Comment (al-kau) [inline source/TuviAuthProtonLib/Broker.cs:87]: Here I would leave the parameter name 'modulus'

File: source/TuviAuthProtonLib/Services/ISRPClient.cs [hunk 4]
```diff
@@ -25,7 +25,7 @@ public interface ISRPClient
             string username,
             string password,
             string salt,
-            string modulus,
+            string signedMessage,
```
Comment (al-kau) [inline source/TuviAuthProtonLib/Services/ISRPClient.cs:28]: Here I would leave the parameter name 'modulus'

File: source/TuviAuthProtonLib/Services/ProtonSRPClient.cs [hunk 5]
```diff
@@ -44,33 +48,27 @@ public Data.IProof CalculateProof(int version, string username, string password,
                     paramName: nameof(version));
             }
 
-            if (string.IsNullOrEmpty(modulus))
+            if (string.IsNullOrEmpty(signedMessage))
```
Comment (al-kau) [inline source/TuviAuthProtonLib/Services/ProtonSRPClient.cs:51]: Here I would leave the parameter name 'modulus'

File: source/TuviAuthProtonLib/Services/ProtonSRPClient.cs [hunk 6]
```diff
@@ -44,33 +48,27 @@ public Data.IProof CalculateProof(int version, string username, string password,
...
             {
                 throw new AuthProtonArgumentException(
                     message: "Proof could not be calculated. The modulus is required.",
-                    paramName: nameof(modulus));
+                    paramName: nameof(signedMessage));
```
Comment (al-kau) [inline source/TuviAuthProtonLib/Services/ProtonSRPClient.cs:55]: Here I would leave the parameter name 'modulus'

File: source/TuviAuthProtonLib/Services/ProtonSRPClient.cs [hunk 7]
```diff
@@ -44,33 +48,27 @@ public Data.IProof CalculateProof(int version, string username, string password,
...
-            {
-                throw new AuthProtonException("Modulus cannot not be verified.");
-            }
+            string verifiedModulus;
 
-            if (verified.IsValid is false)
+            try
             {
-                throw new AuthProtonException("Invalid modulus.");
+                verifiedModulus = _pgpModule.ReadSignedMessage(signedMessage);
```
Comment (al-kau) [inline source/TuviAuthProtonLib/Services/ProtonSRPClient.cs:62]: Here I would leave the parameter name 'modulus'

`verifiedModulus = _pgpModule.ReadSignedMessage(modulus)`

**Review by ValeraFinebits — APPROVED** [review]

**Review by anatolse — APPROVED** [review]

**Pull request merged**